closed/src/java.base/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -109,50 +130,25 @@ public static final boolean isAlgorithmEnabled(String property, String name) {
      * the given property should be used to control enablement of the given
      * native crypto algorithm.
      *
-     * This method is used for native cryptos that have additional requirements
-     * in order to load.
-     *
      * @param property the property used to control enablement of the given
      *                 algorithm
      * @param name the name of the class or the algorithm
-     * @param satisfied whether the additional requirements are met
-     * @param explanation explanation if the native crypto is not loaded
-     *                    due to the additional requirements not being met
      * @return whether the given native crypto algorithm is enabled
      */
-    public static final boolean isAlgorithmEnabled(String property, String name, boolean satisfied, String explanation) {
+    public static final boolean isAlgorithmEnabled(String property, String name) {
         boolean useNativeAlgorithm = false;
         if (useNativeCrypto) {
             useNativeAlgorithm = Boolean.parseBoolean(
                     GetPropertyAction.privilegedGetProperty(property, "true"));
         }
-        if (useNativeAlgorithm) {
-            /*
-             * User wants to use the native crypto implementation. Ensure that the
-             * native crypto library is loaded successfully. Otherwise, issue a warning
-             * message and fall back to the built-in java crypto implementation.
-             */
-            if (isLoaded()) {
-                if (satisfied) {
-                    if (traceEnabled) {
-                        System.err.println(name + " - using native crypto library.");
-                    }
-                } else {
-                    useNativeAlgorithm = false;
-                    if (traceEnabled) {
-                        System.err.println("Warning: " + name + " native requirements not satisfied. " +
-                                explanation + " Using Java crypto implementation.");
-                    }
-                }
+        /*
+         * User wants to use the native crypto implementation. Ensure that the native crypto library is enabled.
+         * Otherwise, issue a warning message.
+         */
+        if (traceEnabled) {
+            if (useNativeAlgorithm) {
+                System.err.println(name + " - using native crypto library.");
```

Comment:
This message must be changed; it will lead to inaccurate assessments in tests. This only indicates the *intent* to use the native implementations, not that the native implementation *is* being used. For that, the message on line 78 must include "using native crypto library".

Reply:
Moved the message into `loadCryptoLibraries()`.